workflow-templates/im-deploy-on-prem-website.yml

```
@@ -0,0 +1,462 @@
+          demo-value: ''
+          uat-value: ''
+
+      # TODO:  Fill in the iis-server-service-user for WinRm access to the IIS Server and remove the environments that are not used.
+      - name: iis-server-service-user
+        id: iis-server-service-user
+        uses: im-open/set-variable-based-on-environment@v1.0.0
+        with:
+          variable-name: 'IIS_SERVER_SERVICE_USER'
+          current-environment: ${{ needs.set-vars.outputs.ENVIRONMENT }}
+          dev-value: ''
+          qa-value: ''
+          stage-value: ''
+          prod-value: ''
+          demo-value: ''
+          uat-value: ''
+
+      # TODO:  Fill in the iis-server-service-password for WinRm access to the IIS Server and remove the environments that are not used.
+      - name: iis-server-service-password
+        id: iis-server-service-password
+        uses: im-open/set-variable-based-on-environment@v1.0.0
+        with:
+          variable-name: 'IIS_SERVER_SERVICE_PASSWORD'
+          current-environment: ${{ needs.set-vars.outputs.ENVIRONMENT }}
+          dev-value: ''
+          qa-value: ''
+          stage-value: ''
+          prod-value: ''
+          demo-value: ''
+          uat-value: ''
+
+      # TODO:  Fill in the iis-server-cert-path for WinRm access to the IIS Server and remove the environments that are not used.
+      - name: iis-server-cert-path
+        id: iis-server-cert-path
+        uses: im-open/set-variable-based-on-environment@v1.0.0
+        with:
+          variable-name: 'IIS_SERVER_CERT_PATH'
+          current-environment: ${{ needs.set-vars.outputs.ENVIRONMENT }}
+          dev-value: ''
+          qa-value: ''
+          stage-value: ''
+          prod-value: ''
+          demo-value: ''
+          uat-value: ''
+
+  # Each env has their own stakeholder approval environment.  If no required reviewers are set for
+  # that environment, the workflow will continue without requiring anyone to approve the deployment.
+  stakeholder-approval:
+    needs: [set-vars]
+    runs-on: ubuntu-latest # Force this to run on github-hosted runner by using a tag that does not exist on self-hosted runners
+    environment: '${{ needs.set-vars.outputs.ENVIRONMENT }} Stakeholder Approval'
+    steps:
+      - run: echo "The current environment is ${{ needs.set-vars.outputs.ENVIRONMENT }}"
+
+      - name: Approval Received
+        run: echo "Stakeholder approval was received"
+
+  # This job needs to run for all environments because build-deploy relies
+  # on it but the steps inside this job will only run for the Prod env.
+  validate-tag-for-deploys:
+    needs: [set-vars, stakeholder-approval]
+    runs-on: ubuntu-latest # Force this to run on github-hosted runner by using a tag that does not exist on self-hosted runners
+    steps:
+      - run: echo "The current environment is ${{ needs.set-vars.outputs.ENVIRONMENT }}"
+
+      # In this job, always checkout the default branch (not the branch that was provided as an input).  Also use
```

Pre-built artifact change;
```suggestion
      # In this job, always checkout the default branch (not the tag that was provided as an input).  Also use
```